README.md

```diff
@@ -3,148 +3,193 @@
+Plankton can be run directly using Python 2.7 or 3.x, or using a prepackaged Docker image that includes all dependencies. See relevant usage sections for details.
 
-# Chopper simulation
+Resources:
+- [GitHub](https://github.com/DMSC-Instrument-Data/plankton)
+- [DockerHub](https://hub.docker.com/r/dmscid/plankton/)
+- [Dockerfile](https://github.com/DMSC-Instrument-Data/plankton/blob/master/Dockerfile)
 
-There are two ways of installing and running the chopper simulation. The first option is to run it directly using Python,
-the second is to run it in a [Docker containter](https://www.docker.com/).
 
-## Installation and startup I: Python
+## Purpose and Use Cases
 
-Clone the repository in a location of your choice:
+Plankton is being developed in the context of instrument control at the [ESS](http://europeanspallationsource.se), but it is general enough to be used in many other contexts that require detailed, stateful software simulations of hardware devices.
 
-```
-$ git clone https://github.com/DMSC-Instrument-Data/plankton.git
-```
+We consider a detailed device simulation to be one that can communicate using the same protocol as the real device, and that can very closely approximate real device behaviour in terms of what is seen through this protocol. This includes gradual processes, side-effects and error conditions.
+
+The purpose of Plankton is to provide a common framework to facilitate the development of such simulators. The framework provides a common set of tools and abstracts away protocol adapters, which helps minimize code replication and allows the developer of a simulated device to focus on capturing device behaviour.
+
+Potential use cases for detailed device simulators include:
+
+- Developing and testing software that interfaces with the device when it is unavailable
+- Testing failure conditions without risking damage to the physical device
+- Automated system and unit tests of software that communicates with the device
+- Perform "dry runs" against test scripts that are to be run on the real device
+
+Using a simulation for the above has the added benefit that, unlike most real devices, a simulation may be sped up / fast-forwarded past any lengthy delays or processes that occur in the device.
+
+
+## Framework Details
+
+The Plankton framework is built around a cycle-based statemachine that drives the device simulation, and shared protocol adapters that separate the communication layer from the simulated device.
+
+#### Cycle-based
+
+By cycle-based we mean that all processing in the framework occurs during "heartbeat" simulation ticks that propagate calls to `process` methods throughout the simulation, along with a Delta T parameter that contains the time that has passed since the last tick. The device simulation is then responsible for updating its state based on how much time has passed and what input has been received during that time.
```

Comment:
As discussed, Δt sounds good to me (with it spelled out in parentheses the first time it shows up).